### ISSUE TITLE
Stats

### ISSUE THREAD

**Issue by dblooman** [swipely/docker-api#250]
With the release of Docker 1.5, container stats are now available.  Any intention for swipely to implement into gem?

**Comment by bfulton** [comment]
Absolutely, and pulls are welcome.

**Comment by DaveBlooman** [comment]
Had a quick look at this today, but i'm not that familiar with Excon.  The response is chunked JSON.  I'm wondering if it best to take the same approach as logs, with a single snapshot of the current stats as well as streaming metrics, or just go with streaming metrics.

**Comment by bfulton** [comment]
Both snapshot and stream sound very useful!

**Comment by tlunter** [comment]
We often use a pattern where we do both already.  The simplest version being the recent `Docker::Image.create` use of [`.response_block`](https://github.com/swipely/docker-api/blob/master/lib/docker/image.rb#L295) here: https://github.com/swipely/docker-api/blob/master/lib/docker/image.rb#L113

**Comment by mikeblum** [comment]
I've started working on this and have added a `stats` method to Docker::Container like this:
```ruby
  def stats(opts = {}, &block)
    connection.get(path_for(:stats), opts, response_block: block)
  end
```

Here's my example call that works in my docker environment:
```
irb(main):001:0> Docker.url='unix:///docker.sock'
=> "unix:///docker.sock"
irb(main):002:0> Docker::Container.all()[1].stats() { |stream, chunk| puts "#{stream}: #{chunk}" }
{"read":"2016-02-19T04:25:06.628635945Z","precpu_stats":{"cpu_usage":{"total_usage":0,"percpu_usage":null,"usage_in_kernelmode":0,"usage_in_usermode":0},"system_cpu_usage":0,"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}},"cpu_stats":{"cpu_usage":{"total_usage":45180377170,"percpu_usage":[45180377170],"usage_in_kernelmode":16760000000,"usage_in_usermode":6680000000},"system_cpu_usage":125973700000000,"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}},"memory_stats":{"usage":425984,"max_usage":8257536,"stats":{"active_anon":106496,"active_file":81920,"cache":98304,"hierarchical_memory_limit":9223372036854771712,"hierarchical_memsw_limit":9223372036854771712,"inactive_anon":98304,"inactive_file":0,"mapped_file":0,"pgfault":4798931,"pgmajfault":3,"pgpgin":2022941,"pgpgout":2022868,"rss":200704,"rss_huge":0,"swap":0,"total_active_anon":106496,"total_active_file":81920,"total_cache":98304,"total_inactive_anon":98304,"total_inactive_file":0,"total_mapped_file":0,"total_pgfault":4798931,"total_pgmajfault":3,"total_pgpgin":2022941,"total_pgpgout":2022868,"total_rss":200704,"total_rss_huge":0,"total_swap":0,"total_unevictable":0,"total_writeback":0,"unevictable":0,"writeback":0},"failcnt":0,"limit":1044574208},"blkio_stats":{"io_service_bytes_recursive":[{"major":8,"minor":0,"op":"Read","value":86016},{"major":8,"minor":0,"op":"Write","value":0},{"major":8,"minor":0,"op":"Sync","value":0},{"major":8,"minor":0,"op":"Async","value":86016},{"major":8,"minor":0,"op":"Total","value":86016}],"io_serviced_recursive":[{"major":8,"minor":0,"op":"Read","value":7},{"major":8,"minor":0,"op":"Write","value":0},{"major":8,"minor":0,"op":"Sync","value":0},{"major":8,"minor":0,"op":"Async","value":7},{"major":8,"minor":0,"op":"Total","value":7}],"io_queue_recursive":[],"io_service_time_recursive":[],"io_wait_time_recursive":[],"io_merged_recursive":[],"io_time_recursive":[],"sectors_recursive":[]},"pids_stats":{},"network":{"rx_bytes":46492,"rx_packets":840,"rx_errors":0,"rx_dropped":0,"tx_bytes":648,"tx_packets":8,"tx_errors":0,"tx_dropped":0}}
```

I've shamelessly based it off of logs and streaming_logs but can't submit a PR as the unit test is just hanging forever.Here is my branch and fork:

[docker-api fork](https://github.com/mikeblum/docker-api.git) with a branch called `stats_tests` This first pass is just looking at the stream implementation of stats. Any thoughts on why stats behaves differently from streaming_logs? I figured they should act about the same way - to such a degree that the unit test should be largely the same... apparently not...

**Comment by tlunter** [comment]
@randywallace has added the non-streaming version of the stats API. Released in v1.29.0
